File: Backend/src/main/java/com/app/logistica/services/RouteService.java
package com.app.logistica.services;

import com.app.logistica.dtos.route.RouteRequest;
import com.app.logistica.dtos.route.RouteResponse;

import java.util.List;

public interface RouteService {

    RouteResponse createRoute(RouteRequest dto);

    RouteResponse updateRoute(Long id, RouteRequest dto);

    RouteResponse getRoute(Long id);

    List<RouteResponse> getAllRoutes();

    String deleteRoute(Long id);
}

File: ReactWeb/src/services/RouteService.js
import api from "./api.js"

export const getAllRoutes = () => api.get("/routes");
export const getRouteById = (id) => api.get(`/routes/${id}`);
export const createRoute = (route) => api.post("/routes", route);
export const updateRoute = (id, route) => api.put(`/routes/${id}`, route);
export const deleteRoute = (id) => api.delete(`/routes/${id}`);
